cpp/src/text/generate_ngrams.cu

```diff
@@ -165,6 +166,48 @@ std::unique_ptr<cudf::column> generate_ngrams(cudf::strings_column_view const& s
 namespace detail {
 namespace {
 
+constexpr cudf::thread_index_type block_size       = 256;
+constexpr cudf::thread_index_type bytes_per_thread = 4;
+
+/**
+ * @brief Counts the number of ngrams in each row of the given strings column
+ *
+ * Each warp processes a single string.
+ * Formula is `count = max(0,str.length() - ngrams + 1)`
+ * If a string has less than ngrams characters, its count is 0.
+ */
+CUDF_KERNEL void count_char_ngrams_kernel(cudf::column_device_view const d_strings,
+                                          cudf::size_type ngrams,
+                                          cudf::size_type* d_counts)
+{
+  auto const idx = cudf::detail::grid_1d::global_thread_id();
+  if (idx >= (static_cast<cudf::thread_index_type>(d_strings.size()) * cudf::detail::warp_size)) {
+    return;
+  }
```

Comment:
Should this kernel use a grid-stride loop?

Reply:
Optional: If not too much overhead, we can consider using cooperative groups (cg) instead of cub and some other explicit computations to simplify the kernel a bit more. [cg::reduce](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#group-collectives) over (warp/block sized) groups can replace `cub::warp_reduce` without needing TempStorage as well.